tmb-user-mgt/tmb-userdata-impl/src/main/java/de/interseroh/tmb/user/client/UserInformationServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.Window;
+import de.interseroh.tmb.user.client.domain.UserInfoClient;
+import org.fusesource.restygwt.client.*;
+import org.gwtbootstrap3.client.ui.AnchorButton;
+import org.gwtbootstrap3.client.ui.base.ComplexWidget;
+import org.gwtbootstrap3.client.ui.constants.ButtonType;
+
+import java.util.logging.Logger;
+
+public class UserInformationServiceImpl implements UserInformationService {
+
+    public final String OID_CONNECT_GATEWAY_LOCATION;
+    public final String USER_INFO_URL;
+    public final String COOKIE_PATH;
+    public final String LOGOUT_URL;
+
+    private static final Logger logger = Logger
+            .getLogger(UserInformationServiceImpl.class.getName());
+
+    public UserInformationServiceImpl(String gatewayLocation, String userInfoUrl, String cookiePath, String logoutUrl) {
+        OID_CONNECT_GATEWAY_LOCATION=gatewayLocation;
+        USER_INFO_URL = userInfoUrl;
+        COOKIE_PATH = cookiePath;
+        LOGOUT_URL = logoutUrl;
+    }
+
+    public UserInformationServiceImpl(UserInformationServiceImpl dolly){
+        this.OID_CONNECT_GATEWAY_LOCATION = dolly.OID_CONNECT_GATEWAY_LOCATION;
+        this.USER_INFO_URL = dolly.USER_INFO_URL;
+        this.COOKIE_PATH = dolly.COOKIE_PATH;
+        this.LOGOUT_URL = dolly.LOGOUT_URL;
+    }
+
+    @Override
+    public ComplexWidget createLoginButton() {
+        AnchorButton loginButton = new AnchorButton(ButtonType.fromStyleName("fa-user"));
+        loginButton.getElement().addClassName("userLogin");
+        loginButton.setHref(OID_CONNECT_GATEWAY_LOCATION);
+        loginButton.setId(ID_LOGIN_BUTTON);
+
+        return loginButton;
+    }
+
+    @Override
+    public ComplexWidget createLogoutButton(final Callback logoutCallback) {
+        final UserInformationServiceImpl me = this;
+        AnchorButton logoutButton =  new AnchorButton(ButtonType.fromStyleName("fa-user"));
+        logoutButton.getElement().addClassName("userLogin");
+        logoutButton.setId(ID_LOGOUT_BUTTON);
+        logoutButton.addClickHandler(new ClickHandler() {
+            final UserInformationServiceImpl userService = new UserInformationServiceImpl(me);
+            @Override
+            public void onClick(ClickEvent event) {
+                try {
+                    userService.logger.info("PERFORM LOGOUT");
+                    userService.performLogout();
+                    logoutCallback.onSuccess(null);
+                    if (LOGOUT_URL != null && !LOGOUT_URL.trim().isEmpty()) {
+                        userService.logger.info("MOVING TO "+LOGOUT_URL);
+                        Window.Location.replace(LOGOUT_URL);
+                    }
+                } catch (Exception e){
+                    logoutCallback.onFailure(e);
+                }
+
+            }
+        });
+        return logoutButton;
+    }
+
+    public boolean performLogout() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1161)